### DISCUSSION TITLE
Integration with an ActivityPub messaging adapter

### DISCUSSION THREAD

**Discussion by aschrijver** [m-ld/m-ld-spec#67] [Ideas]
@gsvarovsky would m-ld be operable in the context of the Fediverse with an ActivityPub messaging adapter?

See also: https://socialhub.activitypub.rocks/t/querying-activitypub-collections/1866/5

**Comment by gsvarovsky** [comment]
Thanks for opening the ticket @aschrijver. Can you point me in the right direction to find out more about how ActivityPub 'does' messaging?

When I look into the docs it seems to be client-server over HTTP, with standardised payloads.

I haven't found anything about pubsub or peer-to-peer, are these considered?

**Comment by aschrijver** [comment]
Besides the [ActivityPub](https://www.w3.org/TR/activitypub/) recommendation, which is based on [ActivityStreams](https://www.w3.org/TR/activitystreams-vocabulary) (the message types, defining 'social primitives', where AS/AP allows extending with custom AP vocabulary extensions), there's the [Guide for new ActivityPub implementers](https://socialhub.activitypub.rocks/t/guide-for-new-activitypub-implementers/479) with a ton of links.

I don't know about PubSub specifically, but research wrt future p2p cabilities (in a hybrid federated + p2p Fediverse) is done in the [DREAM](https://dream.public.cat) community (via e.g. 'content addressing' to get rid of DNS domain registrar dependence). (Regarding the exciting fedi future there's also [The Spritely Project](https://socialhub.activitypub.rocks/t/spritely-project-the-super-exciting-future-of-fediverse/928))
